packages/core/content-manager/server/src/controllers/collection-types.ts

```diff
@@ -390,14 +390,39 @@ export default {
         .countRelations()
         .build();
 
-      const document = id
-        ? await updateDocument(ctx, { populate })
-        : await createDocument(ctx, { populate });
+      const document = await documentManager.findOne(id, model, { populate });
 
       if (permissionChecker.cannot.publish(document)) {
         throw new errors.ForbiddenError();
       }
 
+      /**
+       * Publish can be called on two scenarios:
+       * 1. Create a new document and publish it in one request
+       * 2. Update an existing document and publish it in one request
+       *
+       * Based on user permissions:
+       * 1. User cannot create a document, but can publish
+       *    Action will be forbidden as user cannot create a document
+       * 2. User can update and publish a document
+       *    Action will be allowed, but document will not be updated, only published with the latest draft
+       */
+      const isCreate = !id;
+      const isUpdate = id;
```

Comment:
WHAT IF `id` is `0`? :eyes: 
(it shouldn't happen but maybe being more accurate in our checks could prevent bugs in weird setups in the future?)

Reply:
Also, `isXXX` often reference boolean, but the inference will probably won't consider your variables as such, you might want to enforce some type here

Reply:
good catch! will update that 🚀